warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,247 @@
+     * </p>
+     *
+     * @param config
+     *            the Hadoop Configuration containing whindex rules.
+     * @throws IllegalArgumentException
+     *             if there are configuration issues.
+     */
+    @Override
+    public void setup(Configuration config) throws IllegalArgumentException {
+        // Construct a common prefix based on the type name and the whindex rules constant.
+        String commonPrefix = type.typeName() + "." + WHINDEX_RULES + ".";
+
+        // Get all properties from the configuration that start with the common prefix.
+        Map<String,String> properties = config.getPropsWithPrefix(commonPrefix);
+        // Map to temporarily group properties by their group ID.
+        Map<String,WhindexConfig> groupingsToConfigs = new HashMap<>();
+
+        // Process each configuration property.
+        for (Map.Entry<String,String> entry : properties.entrySet()) {
+            // Split the key by '.' to extract the group ID and property name.
+            String[] parts = entry.getKey().split("\\.");
+            String groupID = parts[0];
+            String property = parts[1];
+
+            // Retrieve or create the WhindexConfig for the given group ID.
+            WhindexConfig whindexConfig = groupingsToConfigs.computeIfAbsent(groupID, (k) -> new WhindexConfig());
+            // Set the appropriate property in the WhindexConfig based on the property name.
+            switch (property) {
+                case VALUE_FIELD:
+                    whindexConfig.setValueField(entry.getValue());
+                    break;
+                case SRC_FIELD:
+                    whindexConfig.setSourceField(entry.getValue());
+                    break;
+                case DELETE_SRC_FIELD:
+                    whindexConfig.setOverloaded(Boolean.parseBoolean(entry.getValue()));
+                    // If the configuration specifies deletion of the source field, add it to overloadedFields.
+                    if (whindexConfig.isOverloaded()) {
+                        overloadedFields.add(whindexConfig.getSourceField());
+                    }
+                    break;
+                case DST_FIELD:
+                    whindexConfig.setDestField(entry.getValue());
+                    // Add the destination (whindex) field to the set of destinationFields.
+                    destinationFields.add(entry.getValue());
+                    break;
+                case VALUES:
+                    // Split the comma-separated list of values and set them.
+                    whindexConfig.setValues(List.of(entry.getValue().split(",")));
+                    break;
+                default:
+                    // Log a warning for any unexpected property found in the configuration.
+                    String originalProperty = commonPrefix + groupID + "." + property;
+                    log.warn("Unexpected whindex property given:" + originalProperty + "=" + entry.getValue());
```

Comment:
I think this should be an error, and throw an exception.  If a property was specified that is unexpected, then it was an error/typo that needs to be fixed.